fission-web-server/library/Fission/Web/Server/Types.hs

```diff
@@ -140,6 +155,106 @@ instance MonadDB (Transaction Server) Server where
 instance MonadRelayStore Server where
   getStoreVar = asks linkRelayStoreVar
 
+instance MonadPowerDNS Server where
+  set recType url zoneTxt contents ttl = do
+    logDebug $ mconcat
+      [ "Set DNS "
+      , displayShow recType
+      , " Record "
+      , displayShow url
+      , " to "
+      , displayShow contents
+      ]
+    clientEnv <- pdnsClientEnv
+    let
+      rrset =
+        RRSet { rrset_name = Text.append (textDisplay url) "."
+              , rrset_type = recType
+              , rrset_ttl = ttl
+              , rrset_changetype = Just Replace
+              , rrset_records = Just $ formatRecords contents
+              , rrset_comments = Nothing
+              }
+      rrsets = RRSets [rrset]
+    resp <- liftIO $ runClientM (PDNS.updateRecords "localhost" zoneTxt rrsets) clientEnv
+    case resp of
```

Comment:
Fine as-is, but another phrasing when you're going to use the result immediately:

```haskell
liftIO (runClientM (PDNS.updateRecords "localhost" zoneTxt rrsets) clientEnv) >>= \case
  Left err -> -- ...
```